get-latest-version.py

```diff
@@ -35,58 +38,154 @@
+        "user.email",
+        "github-actions[bot]@users.noreply.github.com",
+    ],
+    check=False,
+)
+subprocess.run(
+    ["git", "config", "--global", "user.name", "github-actions[bot]"], check=False
+)
+subprocess.run(
+    [
+        "git",
+        "config",
+        "--global",
+        "--add",
+        "safe.directory",
+        "/__w/os-patches/os-patches",
+    ],
+    check=False,
+)
+
+
+def github_pull_exists(title):
+    """Method for checking if GitHub Actions has already opened an PR with this title"""
+    open_pulls = repo.get_pulls(state="open")
+    for open_pull in open_pulls:
+        if open_pull.title == title and open_pull.user.login == "github-actions[bot]":
             return True
     return False
 
-# Get the current version of a package in elementary os patches PPA
-patched_sources = patches_archive.getPublishedSources(exact_match=True,
-    source_name=component_name,
-    status="Published",
-    distro_series=series)
-if len(patched_sources) == 0:
-    issue_title = "Package `%s` not found in os-patches PPA" % (component_name)
-    if not github_issue_exists(issue_title):
-        issue = repo.create_issue(issue_title, "`%s` found in the import list, but not in the PPA. Not deployed yet or removed by accident?" % (component_name))
-        print("Package `%s` not found in elementary os-patches! - Created issue %d" % (component_name, issue.number))
-    sys.exit(0)
 
-patched_version = patched_sources[0].source_package_version
+def get_patched_sources():
+    """Get the current version of a package in elementary os patches PPA"""
+    return patches_archive.getPublishedSources(
+        exact_match=True,
+        source_name=component_name,
+        status="Published",
+        distro_series=series,
+    )
 
-# Search for a new version in the Ubuntu repositories
-pockets = ["Release", "Security", "Updates"]
-for pocket in pockets:
-    found_sources = ubuntu_archive.getPublishedSources(exact_match=True,
+
+def get_upstream_sources():
+    """Get the current version of a package in upstream PPA"""
+    return ubuntu_archive.getPublishedSources(
+        exact_match=True,
         source_name=component_name,
         status="Published",
         pocket=pocket,
-        distro_series=upstream_series)
-    if len(found_sources) > 0:
-        pocket_version = found_sources[0].source_package_version
+        distro_series=upstream_series,
+    )
+
+
+if len(get_patched_sources()) == 0:
+    issue_title = f"Package {component_name} not found in os-patches PPA"
+    if not github_pull_exists(issue_title):
+        issue = repo.create_issue(
+            issue_title,
+            f"{component_name} found in the import list, but not in the PPA. Not deployed yet or removed by accident?",
+        )
+        print(
+            f"Package {component_name} not found in elementary os-patches! - Created issue {issue.number}"
+        )
+    sys.exit(0)
+
+patched_version = get_patched_sources()[0].source_package_version
```

Comment:
I think we should store the result of `get_patched_sources()` into some variable and refer it instead of running it multiple times.

Reply:
Good catch.